File: solutions/s1.cpp
#include <iostream>
#include <memory>
#include <string>

using namespace std;


void tester(){
    cout << "hey you on tester";
}

void safe() {
    auto buffer = make_unique<string>();
    cout << "Enter some text: ";
    cin >> *buffer;

    cout << "You entered: " << *buffer << endl;
}

int main(){
    safe();
    cout << "Back to main!" << endl;
    return 0;
}

File: solutions/s1.rs
use std::io::{self, Write};

fn vulnerable_function() {
    let mut buffer = String::with_capacity(10);
    print!("Enter something: ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut buffer).unwrap();
    println!("You entered: {}", buffer);
}

fn main() {
    vulnerable_function();
}
